src/providers/wms/qgswmscapabilities.cpp

```diff
@@ -2157,9 +2157,25 @@ const QgsWmtsTileMatrix *QgsWmtsTileMatrixSet::findOtherResolution( double tres,
   QMap<double, QgsWmtsTileMatrix>::const_iterator it = tileMatrices.constFind( tres );
   if ( it == tileMatrices.constEnd() )
     return nullptr;
-  it += offset;
-  if ( it == tileMatrices.constEnd() )
-    return nullptr;
+  while ( 1 )
+  {
+    if ( offset > 0 )
+    {
+      ++it;
+      --offset;
+    }
+    else if ( offset < 0 )
+    {
+      if ( it == tileMatrices.constBegin() )
+        return nullptr;
+      --it;
+      ++offset;
+    }
+    else
+      break;
 
+    if ( it == tileMatrices.constEnd() )
+      return nullptr;
+  }
```

Comment:
I'm curious - what's the change in qt5 which required this? It's hard to tell from this commit.